### PR TITLE
added release notes for patch release 15.0.1

### PR DESCRIPTION
**Pull request by dsawali**
Updates release notes to include patch release v15.0.1

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [x] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [x] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: website/versioned_docs/version-15.0.0/version.md [hunk 1]
```diff
@@ -2,6 +2,35 @@
 title: Versions
 author: Jev Bjorsell
 ---
+# Taquito v15.0.1
```
Comment (zainen) [inline website/versioned_docs/version-15.0.0/version.md:5]: these changes seem a little funny to me 
these are for the version 15.0.0 docs? is that right that this should be updated with version 15.0.1?

Comment (dsawali) [reply website/versioned_docs/version-15.0.0/version.md:5]: yeah, we don't really make a versioned_doc folder to index for patch releases. So I included it in 15.0.0 so it shows our users in both `Next` and current labels

Comment (zainen) [reply website/versioned_docs/version-15.0.0/version.md:5]: 👍

**Review by zainen — APPROVED** [review]
LGTM

**Pull request merged**